client/src/utils/CallAutomationUtils.ts

```diff
@@ -0,0 +1,322 @@
+    },
+    body: JSON.stringify({
+      serverCallId
+    })
+  });
+  if (!response.ok) {
+    console.error('Failed to stop transcription:', response);
+    return false;
+  }
+  console.log('Stopped transcription');
+  return true;
+};
+
+export const connectToCallAutomation = async (callAdaterState: CallAdapterState): Promise<boolean> => {
+  if (callAdaterState.call?.info !== undefined && callAdaterState.call?.state === 'Connected') {
+    const serverCallID = await callAdaterState.call.info.getServerCallId();
+    console.log('Server call ID:', serverCallID);
+    const response = await fetch('/api/connectRoomsCall', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        serverCallId: serverCallID
+      })
+    });
+    if (!response.ok) {
+      throw new Error('Failed to start call with transcription');
+    }
+    return true;
+  }
+  return false;
+};
+
+export const getCallSummaryFromServer = async (
+  adapter: CommonCallAdapter,
+  locale: LocaleCode
+): Promise<SummarizeResult> => {
+  console.log('Getting summary from server...');
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const serverCallId = await adapter.getState().call?.info?.getServerCallId();
+    if (!serverCallId) {
+      console.error('Call ID not found');
+      throw new Error('Call ID not found');
+    }
+
+    const response = await fetch('/api/summarizeTranscript', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ serverCallId: serverCallId, locale: locale })
+    });
+    console.log('/summarizeTranscript response', response);
+
+    if (!response.ok) {
+      alert('Summarization request failed');
+      console.error('Response Failed: ', response.statusText);
+      throw new Error('Summarization request failed');
+    }
+
+    const result = await response.json();
+    console.log('Summary result:', result);
+    return result as SummarizeResult;
+  } catch (error) {
+    console.error('Error fetching summary:', error);
+    throw error;
+  }
+};
+
+/**
+ * updateRemoteParticipants - This function is used to update the remote participants in the call. to be
+ * stored in the server to be used in the transcription and summary.
+ * @param callAdapter - The call adapter instance.
+ */
+export const updateRemoteParticipants = async (
+  remoteParticipants: RemoteParticipant[],
+  callId: string
+): Promise<void> => {
+  if (!remoteParticipants) {
+    console.warn('no remote participants found');
+    return;
+  }
+  const remoteParticipantsDisplayInfo = remoteParticipants.map((participant: RemoteParticipantState) => {
+    if ('communicationUserId' in participant.identifier) {
+      return {
+        communicationUserId: participant.identifier.communicationUserId,
+        displayName: participant.displayName
+      };
+    } else {
+      return {};
```

Comment:
nit: better practice to filter instead of returning an empty object
```
remoteParticipants
  .filter((p) => 'communicationUserId' in participant.identifier)
  .map((p) => ({ communicationUserId: participant.identifier.communicationUserId, displayName: participant.displayName });
```

Reply:
this is fixed/ removed later on. we instead have each user update themselves when they join the call.